cmd/dosa/main.go

```diff
@@ -70,7 +70,8 @@ func (b BuildInfo) Execute(args []string) error {
 type GlobalOptions struct {
 	Host        string     `long:"host" default:"127.0.0.1" description:"The hostname or IP for the gateway."`
 	Port        string     `short:"p" long:"port" default:"21300" description:"The hostname or IP for the gateway."`
-	ServiceName string     `short:"s" long:"service" default:"dosa-gateway" description:"The TChannel service name for the gateway."`
+	ServiceName string     `short:"s" long:"service" default:"dosa-gateway" description:"The service name for the gateway."`
```

Comment:
Can other transports be addressed by name? I'd think it is not relevant for HTTP... (perhaps I'm wrong)

Reply:
In this case we're conflating `CallerName` with `ServiceName` which in my mind implies the destination service. YARPC passes the destination service and destination procedure in all requests, regardless of transport.

After poking around, this may just be a idiosyncrasy specific to tchannel, where we require a service name on outbound creation: 

- Comment from abg: https://github.com/yarpc/yarpc-go/blob/ff8d19f1accd1b94e829b698d205a7258f82ddc9/transport/tchannel/channel_outbound.go#L118
- `NewChannelTransport` comments: https://github.com/yarpc/yarpc-go/blob/6151af6af236bde9aa512c58d57661265ef28522/transport/tchannel/channel_transport.go#L49

I'm not familiar with tchannels design but I'm guessing these are just named bidirectional channels and service name is the key.